src/libs/isReportMessageAttachment.ts

```diff
@@ -13,10 +15,13 @@ export default function isReportMessageAttachment(message: Message | undefined):
         return false;
     }
 
-    if (message.translationKey && message.text === CONST.ATTACHMENT_MESSAGE_TEXT) {
-        return message?.translationKey === CONST.TRANSLATION_KEYS.ATTACHMENT;
+    if (message.translationKey) {
+        return message.text === CONST.ATTACHMENT_MESSAGE_TEXT && message.translationKey === CONST.TRANSLATION_KEYS.ATTACHMENT;
     }
 
-    const regex = new RegExp(` ${CONST.ATTACHMENT_SOURCE_ATTRIBUTE}="(.*)"`, 'i');
-    return (message.text === CONST.ATTACHMENT_MESSAGE_TEXT || !!Str.isVideo(message.text)) && (!!message.html.match(regex) || message.html === CONST.ATTACHMENT_UPLOADING_MESSAGE_HTML);
+    const isAttachmentMessageText = message.text === CONST.ATTACHMENT_MESSAGE_TEXT;
+    const isVideoText = Str.isVideo(message.text);
```

Comment:
can we somehow save calculating `isVideo`? Looking from the traces it costs a lot to run it, maybe there are cheaper checks here that can bail out earlier?

I'd say simply moving `hasAttachmentHtml` to the 1st position can vastly limit this time as it also has to evaluate to `true`. IMO we can continue with `isVideo` inly if this check goes through. What are your thoughts on this?

Reply:
Your suggestion makes sense. By moving the `hasAttachmentHtml` check first, we can avoid running the costly `isVideo` check unless absolutely necessary.

Reply:
right, so:
- `hasAttachmentHtml` can go first, 
- then we can check for `isAttachmentMessageText` and as a last resort calculate `isVideoText` as a fallback for it.

Reply:
Changed,  `isVideo` no longer appears in the trace
<img width="1505" alt="Screenshot 2024-07-19 at 11 13 22" src="https://github.com/user-attachments/assets/4594eaee-e38b-4432-b134-f143e1a39c80">